cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/service/BusinessMetadataStore.java

```diff
@@ -230,6 +336,12 @@ public Void apply(BusinessMdsIterable input) throws Exception {
     });
   }
 
+  private void publish(MetadataRecord previous, MetadataRecord additions, MetadataRecord deletions) {
```

Comment:
this may be called with null, if publish is not enabled. You need to protect against that. Also annotate with @Nullable